### PR TITLE
ALTAPPS-1035: iOS challenges

### PR DESCRIPTION
**Pull request by ivan-magda**
**YouTrack Issues**:
[#ALTAPPS-1035](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-1035)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**

1. Adds `ChallengeWidget` module
2. Integrates `ChallengeWidget` module into the Home

### CODE REVIEW

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/HomeSubmodules/ChallengeWidget/Views/ChallengeWidgetView.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    let viewStateKs: ChallengeWidgetViewStateKs
+
+    let viewModel: ChallengeWidgetViewModel
+
+    var body: some View {
+        buildBody()
+    }
+
+    @ViewBuilder
+    private func buildBody() -> some View {
```
Comment (vladkash) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/HomeSubmodules/ChallengeWidget/Views/ChallengeWidgetView.swift:22]: nit: I don't see reasons to extract `buildBody` function here, `body` property doesn't contain any other code and we can substitute function call to it's body content

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/HomeSubmodules/ChallengeWidget/Views/Content/ProgressGrid/ChallengeWidgetContentStateProgressGridView.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    private var gridView: some View {
+        LazyVGrid(
+            columns: Array(
+                repeating: GridItem(
+                    .flexible(),
+                    spacing: LayoutInsets.smallInset,
+                    alignment: .top
+                ),
+                count: 7
```
Comment (vladkash) [inline iosHyperskillApp/iosHyperskillApp/Sources/Modules/HomeSubmodules/ChallengeWidget/Views/Content/ProgressGrid/ChallengeWidgetContentStateProgressGridView.swift:23]: nit: usually we extract such hardcode numbers to `Appearance` structure

**Review by vladkash — APPROVED** [review]
Great! I especially like the way we use to process links in description of challenge

File: iosHyperskillApp/iosHyperskillApp/Sources/Modules/HomeSubmodules/ChallengeWidget/Views/ChallengeWidgetView.swift (hunk 1 above)
Comment (ivan-magda) [reply iosHyperskillApp/iosHyperskillApp/Sources/Modules/HomeSubmodules/ChallengeWidget/Views/ChallengeWidgetView.swift:22]: >  I don't see reasons to extract buildBody function here

I can't find the resource where I read about this, but this is done in performance reason - it's always better to have a small `body` and extract some views into helper computed properties / functions or own files

**Pull request merged**